docs/walk-through/parameters.md

```diff
@@ -89,3 +89,34 @@ spec:
 ```
 
 In this workflow, both steps `A` and `B` would have the same log-level set to `INFO` and can easily be changed between workflow submissions using the `-p` flag.
+
+If using the UI, the `enum` parameter type may also be used to restrict the parameter to certain values.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: hello-world-enum-parameters-
+spec:
+  entrypoint: whalesay
+  arguments:
+    parameters:
+    - name: message
+      default: hello
+      enum:
```

Comment:
huh I've actually never used this as a top-level Workflow parameter 👀  this use-case is indeed undocumented currently then... so I guess we should document it here as there's no other doc on parameters

Reply:
Oh, this _is_ documented elsewhere; #7114 added this under the `WorkflowTemplate` page. That partly makes sense since top-level parameters would only be available with `WorkflowTemplates` -- if you submit a Workflow directly, you wouldn't have access to the UI selection menu. The walk through actually doesn't mention `WorkflowTemplates` at all, so back to square one of me thinking this doesn't quite fit here

I think one of the problems with the current placement is that it's under "Managing WorkflowTemplates" but this feature is not specific to maintaining them, it is for using them. I see that there's no real other place for `WorkflowTemplate` docs currently though.

Maybe we just link to those docs instead then

Reply:
I was just looking back at how I used `enum` it was in a workflow template, so this example is definitely wrong.

I do think I would have found the docs if they were in the UI section. We could move [this section ](https://github.com/Hadrian-MTV/charts/blob/main/applications/flow/factory-execution/values-prod.yaml#L12)to `UI Features` and add screenshots? And then reference the new page within the `WorkflowTemplate` page.

Reply:
> We could move [this section](https://github.com/Hadrian-MTV/charts/blob/main/applications/flow/factory-execution/values-prod.yaml#L12)

Your link above 404s for me; that seems to be a link to a private repo's Helm values?


> to `UI Features` and add screenshots?

The [Intermediate Parameters](https://argo-workflows.readthedocs.io/en/release-3.5/intermediate-inputs/) page is already under UI Features. Usage with WorkflowTemplates could perhaps be a prelude to that all on one page

Reply:
Sorry! [This](https://argo-workflows.readthedocs.io/en/latest/workflow-templates/#ui) is where I meant to link to.

And that makes sense to me.

Reply:
Ok I thought that was what you meant 😅.
I think we can move that to the Intermediate Parameters page and rename the title there a bit